cpp/arcticdb/version/version_store_api.cpp

```diff
@@ -935,17 +932,28 @@ void PythonVersionStore::delete_all_versions(const StreamId& stream_id) {
         log::version().warn("Symbol: {} does not exist.", stream_id);
         return;
     }
-    auto all_index_keys = version_map()->delete_all_versions(store(), stream_id);
-    ARCTICDB_DEBUG(log::version(), "Version heads deleted for symbol {}. Proceeding with index keys total of {}", stream_id, all_index_keys.size());
-    if (!cfg().write_options().delayed_deletes()) {
-        delete_tree({all_index_keys.begin(), all_index_keys.end()});
-    } else {
-        ARCTICDB_DEBUG(log::version(), "Not deleting data for {}", stream_id);
-    }
 
-    if(cfg().symbol_list())
-        symbol_list().remove_symbol(store(), stream_id);
-    ARCTICDB_DEBUG(log::version(), "Delete of Symbol {} successful", stream_id);
+    try {
+        auto [version_id, all_index_keys] = version_map()->delete_all_versions(store(), stream_id);
+        if (cfg().symbol_list())
+            symbol_list().remove_symbol(store(), stream_id, version_id);
+
+        ARCTICDB_DEBUG(log::version(),
+                       "Version heads deleted for symbol {}. Proceeding with index keys total of {}",
+                       stream_id,
+                       all_index_keys.size());
+        if (!cfg().write_options().delayed_deletes()) {
+            delete_tree({all_index_keys.begin(), all_index_keys.end()});
+        } else {
+            ARCTICDB_DEBUG(log::version(), "Not deleting data for {}", stream_id);
+        }
+
+        ARCTICDB_DEBUG(log::version(), "Delete of Symbol {} successful", stream_id);
+    } catch(const StorageException& ex) {
+        log::version().error("Got storage exception in delete - possible parallel deletion?: {}", ex.what());
```

Comment:
There are no actual ill-effects from doing this, the deleted data will still be gone and the rest of the symbol will be readable, I just wanted to advise the user if possible as they are probably doing something inadvisable (for example as you point out if they combine writes with writes or writes with deletes they will get last-writer-wins).

I'm thinking about how we can have more robust concurrenct guarantees over single symbols generally, but there's no shortcuts, it will have to be done in quite a different way.